release: v1.0.1

Xunzhuo opened the pull request:
**What type of PR is this?**
<!--
Your PR title should be descriptive, and generally start with type that contains a subsystem name with `()` if necessary 
and summary followed by a colon. format `chore/docs/feat/fix/refactor/style/test: summary`.
Examples:
* "docs: fix grammar error"
* "feat(translator): add new feature"
* "fix: fix xx bug"
* "chore: change ci & build tools etc"
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->

release: v1.0.1

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,26 @@
+changes:
+  - area: installation
+    change: |
+      Bumped to Envoy v1.29.3
```
```suggestion
      Updated EnvoyProxy version to v1.29.3
```

Xunzhuo replied:
Done

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,26 @@
+  - area: installation
+    change: |
+      Updated EnvoyProxy version to v1.29.3
+      Fixed Run certgen when upgrading
```
```suggestion
      Fixed certgen to support creating the hmac secret during an upgrade
```

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,26 @@
+  - area: infra-manager
+    change: |
+      Fixed do not create infra resources when missing translated listeners
+      Fixed use Patch API for infra-client
```
we never brought the Patch fix in, lets rm it

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,25 @@
+
+  - area: infra-manager
+    change: |
+      Fixed do not create infra resources when missing translated listeners
```
```suggestion
      Skip creating infra resources when the InfraIR has empty listeners
```

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,25 @@
+      Fixed Change route sorting order to Exact > RegularExpression > PathPrefix
+      Fixed infraIR duplicate port translation for merged gateways
+      Fixed set SpawnUpstreamSpan to true
+      Fixed rate limit doesn't work with multiple listeners
```
```suggestion
      Allow rate limit to work with multiple listeners
```

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,25 @@
+      Fixed set path prefix for http ext auth service
+      Fixed Change route sorting order to Exact > RegularExpression > PathPrefix
+      Fixed infraIR duplicate port translation for merged gateways
+      Fixed set SpawnUpstreamSpan to true
```
```suggestion
      Set SpawnUpstreamSpan to true
```

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,25 @@
+      Fixed double slashes in redirect URL
+      Fixed allow Policy to attach to multiple http listeners
+      Fixed set path prefix for http ext auth service
+      Fixed Change route sorting order to Exact > RegularExpression > PathPrefix
```
```suggestion
      Set the route matching precedence order to Exact > RegularExpression > PathPrefix
```

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,25 @@
+      Fixed allow websockets in url rewrite
+      Fixed set host for http health checker explicitly
+      Fixed double slashes in redirect URL
+      Fixed allow Policy to attach to multiple http listeners
```
```suggestion
      Allow ClientTrafficPolicy to attach to multiple http (non https) listeners within the same Gateway
```

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,25 @@
+      Fixed nil secret in resourceversiontable
+      Fixed add missing http filters to the http filter chain
+      Fixed allow websockets in url rewrite
+      Fixed set host for http health checker explicitly
```
```suggestion
      Set the Host header for http health checker
```

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,25 @@
+    change: |
+      Fixed nil secret in resourceversiontable
+      Fixed add missing http filters to the http filter chain
+      Fixed allow websockets in url rewrite
```
```suggestion
      Allow websockets when url rewrite is enabled
```

arkodg commented on release-notes/v1.0.1.yaml:
```diff
@@ -0,0 +1,25 @@
+  - area: translator
+    change: |
+      Fixed nil secret in resourceversiontable
+      Fixed add missing http filters to the http filter chain
```
```suggestion
      Add missing http filters to the http filter chain
```

arkodg commented:
> can we just reuse doc from v1.0.0?

@zirain can you elaborate ?

zirain commented:
> > can we just reuse doc from v1.0.0?
> 
> @zirain can you elaborate ?

there's little change around doc between v1.0.0 and v1.0.1, so should not create a new version of doc?

zirain commented:
> > > > can we just reuse doc from v1.0.0?
> > > 
> > > 
> > > @zirain can you elaborate ?
> > 
> > 
> > there's little change around doc between v1.0.0 and v1.0.1, so should not create a new version of doc?
> 
> We either need to set docs to v1.0.1 or rename dir to v1.0, I believe former is easier cc @Xunzhuo

At least, should do this later

Xunzhuo commented:
> can we just reuse doc from v1.0.0?

Actually, it is reused from v1.0.0, just changed path and modified the related image tags in quickstart. We do not generate new paths for patch release, if we do so, there will be two paths for v1.0.0 and v1.0.1.

zirain commented on .github/workflows/docs.yaml:
```diff
@@ -40,6 +40,7 @@ jobs:
 
     - name: Check links
       run: make docs docs-check-links
+      continue-on-error: true
```
wait https://github.com/envoyproxy/gateway/pull/3146

zirain replied:
I think you can remove this now.

Xunzhuo commented on site/content/en/v1.0.1/user/quickstart.md:
```diff
@@ -43,7 +43,7 @@ unprivileged port, so that Envoy Gateway doesn't need additional privileges.
 consideration when debugging.
 
-[`quickstart.yaml`]: https://github.com/envoyproxy/gateway/releases/download/v1.0.0/quickstart.yaml
+[`quickstart.yaml`]: https://github.com/envoyproxy/gateway/releases/download/v1.0.1/quickstart.yaml
```
Some links are 404 as expected since release is not cut off yet

zirain replied:
this link will never 404, because github.com is ignored because of rate limiting.
